data/scenarios/02Tutorials/06-scan.yaml

```diff
@@ -0,0 +1,62 @@
+name: Scan
+description: |
+  Learn how to send robots to safely scan your surroundings.
+entities:
+  - name: Goal
+    display:
+      attr: device
+      char: 'X'
+    description:
+    - When you land on an alien planet, all the entites in the
+      world will be unfamiliar to you.
+    - But you can learn what they are using the scanner device.
+    - Send robots to move to the unknown entites (marked as '?'),
+     'scan forward' and after that come back and 'upload base'.
+win: |
+  try {
+    bm <- as base {knows "mountain"};
+    bt <- as base {knows "tree"};
+    bw <- as base {knows "wavy water"};
+    return (bm && bt && bw)
+  } { return false }
+solution: |
+  build {
+    turn east; move; move; move;
+    turn left; move; turn right;
+    scan left; move; scan left; scan right;
+    turn back; move; turn left; move;
+    turn west; move; move; move;
+    upload base;
+  }
+robots:
+  - name: base
+    loc: [0,0]
+    dir: [1,0]
+    devices:
+      - logger
+      - 3D printer
+    inventory:
+      - [1, Goal]
+      - [10, logger]
+      - [10, compass]
+      - [10, scanner]
+world:
+  default: [blank, none]
+  palette:
+    '.': [grass, null]
+    '~': [ice, wavy water]
+    'T': [grass, tree]
+    'A': [stone, mountain]
+    '┌': [stone, upper left corner]
+    '┐': [stone, upper right corner]
+    '└': [stone, lower left corner]
+    '┘': [stone, lower right corner]
+    '─': [stone, horizontal wall]
+    '│': [stone, vertical wall]
+  upperleft: [-1, 3]
+  map: |
+    ┌─────┐
+    │AAAT~│
+    │..A..│
+    │....A│
+    └─────┘
```

Comment:
This one is kind of tedious.  I think we could make it a bit smaller and not require quite so much maneuvering around mountains.

Reply:
I put the water closer to save one turn and now you win if you scan *anything* and upload it to base.